Networking/Networking/Extensions/KeyedDecodingContainer+Woo.swift

```diff
@@ -127,6 +127,21 @@ extension KeyedDecodingContainer {
 
         return nil
     }
+
+    /// Decodes an array of the specified Type for a given Key (if present), discarding any malformed items in the array.
+    ///
+    /// **WARNING:** Only use this if it's acceptable to handle and display partial data!
+    ///
+    /// This method *does NOT throw*. We want this behavior so that if a malformed entity is received, we just skip it, rather
+    /// than breaking the entire parsing chain.
+    ///
+    func failsafeDecodeIfPresent<T>(lossyList: [T].Type, forKey key: KeyedDecodingContainer<K>.Key) -> [T] where T: Decodable {
+        do {
+            return try decodeIfPresent(LossyDecodableList<T>.self, forKey: key)?.elements ?? []
```

Comment:
Hm, I'm probably missing something but I don't understand how to iterate over the array here until after it's decoded. (And in that case it will throw if it encounters an invalid item.) My understanding is that we need this `LossyDecodableList` type to essentially make each item discardable when it's decoded. Is there another way to iterate over the array?